legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -186,3 +192,40 @@ def validate_share_structure(incorporation_json) -> Error:
         return msg
 
     return None
+
+
+def validate_incorporation_effective_date(incorporation_json) -> Error:
+    """Return an error or warning message based on the effective date validation rules.
+
+    Rules:
+        - The effective date must be the correct format.
+        - The effective date must be a minimum of 2 minutes in the future.
+        - The effective date must be a maximum of 10 days in the future.
+    """
+    # Setup
+    msg = []
+    now = dt.utcnow()
+    now_plus_2_minutes = now + timedelta(minutes=2)
```

Comment:
Is this working fine if you choose exactly 2 mins in the UI?

Reply:
The front end won't let you File And Pay if it is any sooner than 2 minutes, yea.

Reply:
For example, if you select exactly 2 minutes ahead, but take 5 minutes to file. You will have to update that time before filing and paying.

Reply:
utcnow will always be ahead of the submission time by milliseconds or so due to network latency and stuff right? I think there is achance that exact 2 mins from the UI may cause a validation failure. Could you please test it out from UI

Reply:
Good point for sure, i know this is an extreme edge case ( someone paying $100 to have it effective in 2 mins from now 😆 ), so should we adjust the time and give a buffer on the back end. It could potentially pass front end validation and then fail the validator during the request.

The front end will not make the request if it is < 2 mins, but you're right, we should take into account the latency, would adding an extra minute be acceptable?

Reply:
Also, this particular one need not block the PR as such and the fix (if any) can be a separate PR

Reply:
Absolutely, will look into this, thanks Lekshmi!